File: src/main/java/org/capturecoop/cclogger/CCLogMessage.java
package org.capturecoop.cclogger;

import java.time.LocalDateTime;

public class CCLogMessage {
    private final CCLogLevel level;
    private final String message;
    private final LocalDateTime time;
    private final StackTraceElement stackTraceElement;
    private final boolean isStacktrace;

    public CCLogMessage(CCLogLevel level, String message, LocalDateTime time, StackTraceElement stackTraceElement, boolean isException) {
        this.level = level;
        this.message = message;
        this.time = time;
        this.stackTraceElement = stackTraceElement;
        this.isStacktrace = isException;
    }

    public CCLogLevel getLevel() {
        return level;
    }

    public String getMessage() {
        return message;
    }

    public LocalDateTime getTime() {
        return time;
    }

    public StackTraceElement getStackTraceElement() {
        return stackTraceElement;
    }

    public boolean isStacktrace() {
        return isStacktrace;
    }
}

File: src/main/kotlin/org/capturecoop/cclogger/CCLogMessage.kt
package org.capturecoop.cclogger

import java.awt.Color
import java.time.LocalDateTime

data class CCLogMessage(
    val level: CCLogLevel,
    val message: String,
    val time: LocalDateTime,
    val stackTraceElement: StackTraceElement?,
    val isException: Boolean
)

enum class CCLogFilter {
    NONE, //Display nothing
    ERRORS, //Display errors
    WARNINGS, //Display warnings and errors
    INFO, //Display info, warnings and errors
    DEBUG //Display everything + debug messages that provide extra info
}

enum class CCLogLevel(private val color: Color) {
    DEBUG(Color.MAGENTA),
    INFO(Color.WHITE),
    WARNING(Color.YELLOW),
    ERROR(Color.RED);
    fun getAnsiColor() = "\u001B[38;2;${color.red};${color.green};${color.blue}m"
    fun getHTMLColor() = "rgb(${color.red}, ${color.green}, ${color.blue})"
    fun getAnsiReset() = "\u001B[0m"
    fun isNot(vararg levels: CCLogLevel): Boolean {
        levels.forEach { if(it == this) return false }
        return true
    }
}